src/platform/packages/shared/response-ops/alerts-table/components/alert_detail_flyout.tsx

```diff
@@ -0,0 +1,181 @@
+    () => ({
+      id: 'overview',
+      'data-test-subj': 'alertFlyoutOverviewTab',
+      name: i18n.ALERT_FLYOUT_OVERVIEW_TAB_TITLE,
+      content: alert && (
+        <EuiPanel hasShadow={false} data-test-subj="alertFlyoutOverviewTabPanel">
+          <EuiDescriptionList
+            listItems={(columns ?? defaultAlertsTableColumns).map((column) => {
+              const value = (alert[column.id] as JsonValue[])?.[0];
+
+              return {
+                title: (column.displayAsText as string) ?? column.id,
+                description:
+                  value != null ? (
+                    <DefaultCellValue
+                      columnId={column.id}
+                      alert={alert}
+                      openLinksInNewTab={openLinksInNewTab}
+                    />
+                  ) : (
+                    '—'
+                  ),
+              };
+            })}
+            type="column"
+            columnWidths={[1, 3]}
+          />
+        </EuiPanel>
+      ),
+    }),
+    [alert, columns, openLinksInNewTab]
+  );
+
+  const tableTab = useMemo(
+    () => ({
+      id: 'table',
+      'data-test-subj': 'alertFlyoutTableTab',
+      name: i18n.ALERT_FLYOUT_TABLE_TAB_TITLE,
+      content: alert && (
+        <EuiPanel hasShadow={false} data-test-subj="alertFlyoutTableTabPanel">
+          <AlertFieldsTable alert={alert} />
+        </EuiPanel>
+      ),
+    }),
+    [alert]
+  );
+
+  const tabs = useMemo(() => [overviewTab, tableTab], [overviewTab, tableTab]);
+  const [selectedTabId, setSelectedTabId] = useState<TabId>('overview');
+
+  const selectedTab = useMemo(
+    () => tabs.find((tab) => tab.id === selectedTabId) ?? tabs[0],
+    [tabs, selectedTabId]
+  );
+
+  return (
+    <EuiFlyout
+      onClose={() => {
+        onExpandedAlertIndexChange?.(null);
+      }}
+      size="m"
+      data-test-subj="alertFlyout"
+      aria-label={ariaLabel}
+      ownFocus={ownFocus}
+    >
+      <EuiFlyoutHeader hasBorder>
+        <EuiTitle size="s">
+          <h3>
+            {!isLoading && alert
+              ? (alert[ALERT_RULE_NAME]?.[0] as string)
+              : i18n.ALERT_FLYOUT_DEFAULT_TITLE}
+          </h3>
+        </EuiTitle>
+        {Boolean(hasPagination) && (
+          <>
+            <EuiSpacer size="m" />
+            <EuiFlexGroup gutterSize="none" justifyContent="flexEnd">
+              <EuiFlexItem grow={false}>
+                <EuiPagination
+                  aria-label={i18n.ALERT_FLYOUT_PAGINATION_ARIA_LABEL}
+                  pageCount={alertsCount}
+                  activePage={expandedAlertIndex}
+                  onPageClick={(activePage) => {
+                    onExpandedAlertIndexChange?.(activePage);
+                  }}
+                  compressed
+                  data-test-subj="alertFlyoutPagination"
+                />
+              </EuiFlexItem>
+            </EuiFlexGroup>
+          </>
+        )}
+      </EuiFlyoutHeader>
+
+      {isLoading ? (
```

Comment:
Before we were showing stale data while a new page of alerts was loading, which was confusing since the navigation page wasn't matching the data shown in the flyout. This ensures that when the current alert is part of a page that's still loading, we show a loading indicator instead